lib/rubocop/cop/lint/redundant_with_index.rb

```diff
@@ -69,7 +69,7 @@ def message(node)
         end
 
         def with_index_range(send)
-          range_between(send.loc.selector.begin_pos, send.loc.selector.end_pos)
+          range_between(send.loc.selector.begin_pos, send.source.length)
```

Comment:
This actually only works incidentally because `processed_source` happens to start at the beginning of the node, which feels kind of frail. Would probably use `send.loc.expression.end_pos` instead of `send.source.length`.

Reply:
I opened a PR #4983. Thanks!